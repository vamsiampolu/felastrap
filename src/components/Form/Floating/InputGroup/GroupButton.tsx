import { GroupButton, GroupButtonProps } from "../../InputGroup/GroupButton";
import { useFela } from "react-fela";

const floatingRule = () => ({
  height: "calc(3.5rem + 2px)",
  minHeight: "calc(3.5rem + 2px)",
});

export const FloatingGroupButton = ({
  className = "",
  children,
  ...props
}: GroupButtonProps<"button">) => {
  const { css } = useFela();
  return (
    <GroupButton
      className={`${css(
        floatingRule,
      )} floating-input-group-button ${className}`.trim()}
      {...props}
    >
      {children}
    </GroupButton>
  );
};
