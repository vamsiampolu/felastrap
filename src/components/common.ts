import type * as Css from "csstype";
import type * as Fela from "fela";
import type {
  ComponentPropsWithoutRef,
  ComponentType,
  ElementType,
} from "react";
import type * as ReactFela from "react-fela";

export type ExcludeFrom<T, U> = T extends U ? never : T;
export type Theme = object;

export type Rule = Fela.TRule;
export type CSSObject = Fela.CSSObject;
export type StyleProps<P = object> = ReactFela.StyleProps<Theme, P>;
export type StyleFunction<P = object> = ReactFela.StyleFunction<Theme, P>;

export type Padding = Required<
  Pick<
    Css.Properties,
    "paddingTop" | "paddingLeft" | "paddingRight" | "paddingBottom"
  >
>;
export type BorderRadius = Required<
  Pick<
    Css.Properties,
    | "borderTopLeftRadius"
    | "borderTopRightRadius"
    | "borderBottomRightRadius"
    | "borderBottomLeftRadius"
  >
>;

export type PaddingString = Required<Css.Properties["padding"]>;
export type Color = Css.Property.Color;

export type HeaderTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
export type Size = "small" | "medium" | "large";
export type Variant =
  | "default"
  | "danger"
  | "info"
  | "primary"
  | "secondary"
  | "success"
  | "warning";

export type FontSize = Css.Property.FontSize;
export type ColorMapping = Record<Variant, Color>;
export type PaddingBySize = Record<Size, Padding>;

export type SpanProps = ComponentPropsWithoutRef<"span">;
export type DivProps = ComponentPropsWithoutRef<"div">;
export type ListItemProps = ComponentPropsWithoutRef<"li">;

export type DropdownDirection = "up" | "down" | "end" | "start" | "centered";
export type CaretDirection = "up" | "down" | "end" | "start";
export type NavVariant =
  | "default"
  | "tabs"
  | "pills"
  | "underline"
  | "vertical";

export type CaretProps = DivProps & {
  width: string | number | null;
};

export type DynamicElement<T extends ElementType = "span"> =
  | keyof JSX.IntrinsicElements
  | ComponentType<T>;
export type PargraphProps = ComponentPropsWithoutRef<"p">;
export type ImageProps = ComponentPropsWithoutRef<"img">;
export type TooltipPlacement = "top" | "left" | "bottom" | "right";

export type MediaQuerySize = "xs" | "sm" | "md" | "lg" | "xl" | "xxl";
export type Timer = ReturnType<typeof setTimeout>;
export type OnClick<T extends ElementType> =
  ComponentPropsWithoutRef<T>["onClick"];
export type StyleKey = Exclude<
  keyof HTMLElement["style"],
  "length" | "parentRule"
>;
