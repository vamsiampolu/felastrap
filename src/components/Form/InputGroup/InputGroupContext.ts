import { createContext } from "react";
import { Size } from "../../common";

export type InputGroupContextValue = {
  size: Size;
  group: boolean;
};

export const InputGroupContext = createContext<InputGroupContextValue>({
  size: "medium",
  group: false,
});
