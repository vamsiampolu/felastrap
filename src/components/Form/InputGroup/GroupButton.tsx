import { useFela } from "react-fela";
import { Button, ButtonProps } from "../../Button";
import { ElementType, useContext } from "react";
import { InputGroupContext } from "./InputGroupContext";
import { Size, StyleProps } from "../../common";

type RuleProps = StyleProps<{ size: Size }>;

const rule = () => ({
  position: "relative",
  zIndex: 2,
  ":focus": {
    zIndex: 5,
  },
});

const largeRule = ({ size }: RuleProps) =>
  size === "large"
    ? {
        padding: "0.5rem 1rem",
        fontSize: "1.25rem",
        borderRadius: "0.5rem",
      }
    : {};

const smallRule = ({ size }: RuleProps) =>
  size === "small"
    ? {
        padding: "0.25rem 0.5rem",
        fontSize: "0.875rem",
        borderRadius: "0.25rem",
      }
    : {};

export type GroupButtonProps<T extends ElementType> = Omit<
  ButtonProps<T>,
  "outline" | "variant" | "size"
>;

export const GroupButton = ({
  className = "",
  children,
  ...props
}: GroupButtonProps<"button">) => {
  const { size } = useContext(InputGroupContext);
  const { css } = useFela({ size });
  const cx = `${css([
    rule,
    smallRule,
    largeRule,
  ])} input-group-btn ${className}`.trim();

  return (
    <Button
      data-testid="input-group-btn"
      outline
      size={size ?? "medium"}
      variant="secondary"
      className={cx}
      {...props}
    >
      {children}
    </Button>
  );
};
