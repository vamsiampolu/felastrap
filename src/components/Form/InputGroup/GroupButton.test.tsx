import { IRenderer } from "fela";
import { GroupButton } from "./GroupButton";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../test-helpers";
import { Size } from "../../common";
import { InputGroupContext, InputGroupContextValue } from "./InputGroupContext";
import { render, screen } from "@testing-library/react";

describe("components/Form/InputGroup/GroupButton", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should display a default input group button", async () => {
    const component = <GroupButton>Group Button</GroupButton>;

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it.each(["small", "medium", "large"])(
    "should display an input group button of size %s",
    async (size) => {
      const value: InputGroupContextValue = {
        size: size as Size,
        group: false,
      };

      const component = (
        <InputGroupContext.Provider value={value}>
          <GroupButton>Group Button</GroupButton>
        </InputGroupContext.Provider>
      );

      render(<Wrapper renderer={renderer}>{component}</Wrapper>);
      expect(screen.getByTestId("input-group-btn")).toBeInTheDocument();

      expect(screen.queryByTestId("input-group-btn")).toHaveClass(
        "input-group-btn",
      );
      expect(screen.queryByTestId("input-group-btn")).toHaveTextContent(
        "Group Button",
      );

      const snapshot = await createSnapshot({ component, renderer });
      expect(snapshot).toMatchSnapshot();
    },
  );
});
