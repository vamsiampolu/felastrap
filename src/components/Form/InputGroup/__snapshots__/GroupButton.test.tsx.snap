// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/InputGroup/GroupButton should display a default input group button 1`] = `
".a {
  position: relative;
}
.b {
  z-index: 2;
}
.d {
  border-top-left-radius: 4px;
}
.e {
  border-top-right-radius: 0;
}
.f {
  border-bottom-right-radius: 0;
}
.g {
  border-bottom-left-radius: 4px;
}
.h {
  padding-top: 6px;
}
.i {
  padding-bottom: 6px;
}
.j {
  padding-left: 12px;
}
.k {
  padding-right: 12px;
}
.l {
  -webkit-background-color: rgba(0, 0, 0, 0);
  background-color: rgba(0, 0, 0, 0);
}
.m {
  -webkit-background-image: none;
  background-image: none;
}
.n {
  color: #6c757d;
}
.o {
  cursor: pointer;
}
.p {
  display: inline-block;
}
.q {
  font-size: 14px;
}
.r {
  font-weight: 400;
}
.s {
  line-height: 1.42857143;
}
.t {
  margin-bottom: 0;
}
.u {
  text-align: center;
}
.v {
  touch-action: manipulation;
}
.w {
  white-space: nowrap;
}
.x {
  vertical-align: middle;
}
.y {
  user-select: none;
}
.ak:disabled {
  -webkit-background-color: transparent;
  background-color: transparent;
}
.al:disabled {
  color: #6c757d;
}
.am:disabled {
  box-shadow: none;
}
.an:disabled {
  cursor: not-allowed;
}
.ao:disabled {
  -webkit-filter: alpha(opacity=65);
  filter: alpha(opacity=65);
}
.ap:disabled {
  opacity: 0.65;
}
.aq:disabled {
  pointer-events: none;
}
.ar:disabled {
  border-top-color: #6c757d;
}
.as:disabled {
  border-right-color: #6c757d;
}
.at:disabled {
  border-bottom-color: #6c757d;
}
.au:disabled {
  border-left-color: #6c757d;
}
.bj {
  gradient: none;
}
.bl {
  border-top-width: 1px;
}
.bm {
  border-right-width: 1px;
}
.bn {
  border-bottom-width: 1px;
}
.bo {
  border-left-width: 1px;
}
.bp {
  border-top-style: solid;
}
.bq {
  border-right-style: solid;
}
.br {
  border-bottom-style: solid;
}
.bs {
  border-left-style: solid;
}
.bt {
  border-top-color: #6c757d;
}
.bu {
  border-right-color: #6c757d;
}
.bv {
  border-bottom-color: #6c757d;
}
.bw {
  border-left-color: #6c757d;
}
.bd:hover {
  -webkit-background-color: #6c757d;
  background-color: #6c757d;
}
.be:hover {
  color: #fff;
}
.bf:hover {
  border-top-color: #6c757d;
}
.bg:hover {
  border-right-color: #6c757d;
}
.bh:hover {
  border-bottom-color: #6c757d;
}
.bi:hover {
  border-left-color: #6c757d;
}
.c:focus {
  z-index: 5;
}
.av:focus-visible {
  color: #fff;
}
.aw:focus-visible {
  -webkit-background-color: #5c636a;
  background-color: #5c636a;
}
.ax:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(130, 138, 145, 0.5);
}
.ay:focus-visible {
  outline-width: 0;
}
.az:focus-visible {
  border-top-color: #565e64;
}
.ba:focus-visible {
  border-right-color: #565e64;
}
.bb:focus-visible {
  border-bottom-color: #565e64;
}
.bc:focus-visible {
  border-left-color: #565e64;
}
.bk:focus {
  box-shadow: 0 0 0 0.25rem rgba(108, 117, 125, 0.5);
}
.z:active {
  color: #fff;
}
.ab:active {
  -webkit-background-color: #6c757d;
  background-color: #6c757d;
}
.ac:active {
  -webkit-background-image: none;
  background-image: none;
}
.ae:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.af:active {
  outline-width: 0;
}
.ag:active {
  border-top-color: #6c757d;
}
.ah:active {
  border-right-color: #6c757d;
}
.ai:active {
  border-bottom-color: #6c757d;
}
.aj:active {
  border-left-color: #6c757d;
}


<button
  data-testid="input-group-btn"
  class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw d e f g h i j k l m n o p q r s t u v w x y z a b c input-group-btn"
>
  Group Button</button
>;
"
`;

exports[`components/Form/InputGroup/GroupButton should display an input group button of size large 1`] = `
".a {
  position: relative;
}
.b {
  z-index: 2;
}
.d {
  font-size: 1.25rem;
}
.e {
  border-top-left-radius: 0.5rem;
}
.f {
  border-top-right-radius: 0.5rem;
}
.g {
  border-bottom-right-radius: 0.5rem;
}
.h {
  border-bottom-left-radius: 0.5rem;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 1rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 1rem;
}
.m {
  border-top-right-radius: 0;
}
.n {
  border-bottom-right-radius: 0;
}
.o {
  background-color: rgba(0, 0, 0, 0);
}
.p {
  background-image: none;
}
.q {
  color: #6c757d;
}
.r {
  cursor: pointer;
}
.s {
  display: inline-block;
}
.t {
  font-weight: 400;
}
.u {
  line-height: 1.42857143;
}
.v {
  margin-bottom: 0;
}
.w {
  text-align: center;
}
.x {
  touch-action: manipulation;
}
.y {
  white-space: nowrap;
}
.z {
  vertical-align: middle;
}
.ab {
  user-select: none;
}
.am:disabled {
  background-color: transparent;
}
.an:disabled {
  color: #6c757d;
}
.ao:disabled {
  box-shadow: none;
}
.ap:disabled {
  cursor: not-allowed;
}
.aq:disabled {
  filter: alpha(opacity=65);
}
.ar:disabled {
  opacity: 0.65;
}
.as:disabled {
  pointer-events: none;
}
.at:disabled {
  border-top-color: #6c757d;
}
.au:disabled {
  border-right-color: #6c757d;
}
.av:disabled {
  border-bottom-color: #6c757d;
}
.aw:disabled {
  border-left-color: #6c757d;
}
.bl {
  gradient: none;
}
.bn {
  border-top-width: 1px;
}
.bo {
  border-right-width: 1px;
}
.bp {
  border-bottom-width: 1px;
}
.bq {
  border-left-width: 1px;
}
.br {
  border-top-style: solid;
}
.bs {
  border-right-style: solid;
}
.bt {
  border-bottom-style: solid;
}
.bu {
  border-left-style: solid;
}
.bv {
  border-top-color: #6c757d;
}
.bw {
  border-right-color: #6c757d;
}
.bx {
  border-bottom-color: #6c757d;
}
.by {
  border-left-color: #6c757d;
}
.bf:hover {
  background-color: #6c757d;
}
.bg:hover {
  color: #fff;
}
.bh:hover {
  border-top-color: #6c757d;
}
.bi:hover {
  border-right-color: #6c757d;
}
.bj:hover {
  border-bottom-color: #6c757d;
}
.bk:hover {
  border-left-color: #6c757d;
}
.c:focus {
  z-index: 5;
}
.ax:focus-visible {
  color: #fff;
}
.ay:focus-visible {
  background-color: #5c636a;
}
.az:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(130, 138, 145, 0.5);
}
.ba:focus-visible {
  outline-width: 0;
}
.bb:focus-visible {
  border-top-color: #565e64;
}
.bc:focus-visible {
  border-right-color: #565e64;
}
.bd:focus-visible {
  border-bottom-color: #565e64;
}
.be:focus-visible {
  border-left-color: #565e64;
}
.bm:focus {
  box-shadow: 0 0 0 0.25rem rgba(108, 117, 125, 0.5);
}
.ac:active {
  color: #fff;
}
.ae:active {
  background-color: #6c757d;
}
.af:active {
  background-image: none;
}
.ag:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.ah:active {
  outline-width: 0;
}
.ai:active {
  border-top-color: #6c757d;
}
.aj:active {
  border-right-color: #6c757d;
}
.ak:active {
  border-bottom-color: #6c757d;
}
.al:active {
  border-left-color: #6c757d;
}


<button
  data-testid="input-group-btn"
  class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by d e h i j k l m n o p q r s t u v w x y z a b c d e f g h i j k l input-group-btn"
>
  Group Button</button
>;
"
`;

exports[`components/Form/InputGroup/GroupButton should display an input group button of size medium 1`] = `
".a {
  position: relative;
}
.b {
  z-index: 2;
}
.d {
  border-top-left-radius: 4px;
}
.e {
  border-top-right-radius: 0;
}
.f {
  border-bottom-right-radius: 0;
}
.g {
  border-bottom-left-radius: 4px;
}
.h {
  padding-top: 6px;
}
.i {
  padding-bottom: 6px;
}
.j {
  padding-left: 12px;
}
.k {
  padding-right: 12px;
}
.l {
  background-color: rgba(0, 0, 0, 0);
}
.m {
  background-image: none;
}
.n {
  color: #6c757d;
}
.o {
  cursor: pointer;
}
.p {
  display: inline-block;
}
.q {
  font-size: 14px;
}
.r {
  font-weight: 400;
}
.s {
  line-height: 1.42857143;
}
.t {
  margin-bottom: 0;
}
.u {
  text-align: center;
}
.v {
  touch-action: manipulation;
}
.w {
  white-space: nowrap;
}
.x {
  vertical-align: middle;
}
.y {
  user-select: none;
}
.ak:disabled {
  background-color: transparent;
}
.al:disabled {
  color: #6c757d;
}
.am:disabled {
  box-shadow: none;
}
.an:disabled {
  cursor: not-allowed;
}
.ao:disabled {
  filter: alpha(opacity=65);
}
.ap:disabled {
  opacity: 0.65;
}
.aq:disabled {
  pointer-events: none;
}
.ar:disabled {
  border-top-color: #6c757d;
}
.as:disabled {
  border-right-color: #6c757d;
}
.at:disabled {
  border-bottom-color: #6c757d;
}
.au:disabled {
  border-left-color: #6c757d;
}
.bj {
  gradient: none;
}
.bl {
  border-top-width: 1px;
}
.bm {
  border-right-width: 1px;
}
.bn {
  border-bottom-width: 1px;
}
.bo {
  border-left-width: 1px;
}
.bp {
  border-top-style: solid;
}
.bq {
  border-right-style: solid;
}
.br {
  border-bottom-style: solid;
}
.bs {
  border-left-style: solid;
}
.bt {
  border-top-color: #6c757d;
}
.bu {
  border-right-color: #6c757d;
}
.bv {
  border-bottom-color: #6c757d;
}
.bw {
  border-left-color: #6c757d;
}
.bd:hover {
  background-color: #6c757d;
}
.be:hover {
  color: #fff;
}
.bf:hover {
  border-top-color: #6c757d;
}
.bg:hover {
  border-right-color: #6c757d;
}
.bh:hover {
  border-bottom-color: #6c757d;
}
.bi:hover {
  border-left-color: #6c757d;
}
.c:focus {
  z-index: 5;
}
.av:focus-visible {
  color: #fff;
}
.aw:focus-visible {
  background-color: #5c636a;
}
.ax:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(130, 138, 145, 0.5);
}
.ay:focus-visible {
  outline-width: 0;
}
.az:focus-visible {
  border-top-color: #565e64;
}
.ba:focus-visible {
  border-right-color: #565e64;
}
.bb:focus-visible {
  border-bottom-color: #565e64;
}
.bc:focus-visible {
  border-left-color: #565e64;
}
.bk:focus {
  box-shadow: 0 0 0 0.25rem rgba(108, 117, 125, 0.5);
}
.z:active {
  color: #fff;
}
.ab:active {
  background-color: #6c757d;
}
.ac:active {
  background-image: none;
}
.ae:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.af:active {
  outline-width: 0;
}
.ag:active {
  border-top-color: #6c757d;
}
.ah:active {
  border-right-color: #6c757d;
}
.ai:active {
  border-bottom-color: #6c757d;
}
.aj:active {
  border-left-color: #6c757d;
}


<button
  data-testid="input-group-btn"
  class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw d e f g h i j k l m n o p q r s t u v w x y z a b c input-group-btn"
>
  Group Button</button
>;
"
`;

exports[`components/Form/InputGroup/GroupButton should display an input group button of size small 1`] = `
".a {
  position: relative;
}
.b {
  z-index: 2;
}
.d {
  font-size: 0.875rem;
}
.e {
  border-top-left-radius: 0.25rem;
}
.f {
  border-top-right-radius: 0.25rem;
}
.g {
  border-bottom-right-radius: 0.25rem;
}
.h {
  border-bottom-left-radius: 0.25rem;
}
.i {
  padding-top: 0.25rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.25rem;
}
.l {
  padding-left: 0.5rem;
}
.m {
  border-top-right-radius: 0;
}
.n {
  border-bottom-right-radius: 0;
}
.o {
  background-color: rgba(0, 0, 0, 0);
}
.p {
  background-image: none;
}
.q {
  color: #6c757d;
}
.r {
  cursor: pointer;
}
.s {
  display: inline-block;
}
.t {
  font-weight: 400;
}
.u {
  line-height: 1.42857143;
}
.v {
  margin-bottom: 0;
}
.w {
  text-align: center;
}
.x {
  touch-action: manipulation;
}
.y {
  white-space: nowrap;
}
.z {
  vertical-align: middle;
}
.ab {
  user-select: none;
}
.am:disabled {
  background-color: transparent;
}
.an:disabled {
  color: #6c757d;
}
.ao:disabled {
  box-shadow: none;
}
.ap:disabled {
  cursor: not-allowed;
}
.aq:disabled {
  filter: alpha(opacity=65);
}
.ar:disabled {
  opacity: 0.65;
}
.as:disabled {
  pointer-events: none;
}
.at:disabled {
  border-top-color: #6c757d;
}
.au:disabled {
  border-right-color: #6c757d;
}
.av:disabled {
  border-bottom-color: #6c757d;
}
.aw:disabled {
  border-left-color: #6c757d;
}
.bl {
  gradient: none;
}
.bn {
  border-top-width: 1px;
}
.bo {
  border-right-width: 1px;
}
.bp {
  border-bottom-width: 1px;
}
.bq {
  border-left-width: 1px;
}
.br {
  border-top-style: solid;
}
.bs {
  border-right-style: solid;
}
.bt {
  border-bottom-style: solid;
}
.bu {
  border-left-style: solid;
}
.bv {
  border-top-color: #6c757d;
}
.bw {
  border-right-color: #6c757d;
}
.bx {
  border-bottom-color: #6c757d;
}
.by {
  border-left-color: #6c757d;
}
.bf:hover {
  background-color: #6c757d;
}
.bg:hover {
  color: #fff;
}
.bh:hover {
  border-top-color: #6c757d;
}
.bi:hover {
  border-right-color: #6c757d;
}
.bj:hover {
  border-bottom-color: #6c757d;
}
.bk:hover {
  border-left-color: #6c757d;
}
.c:focus {
  z-index: 5;
}
.ax:focus-visible {
  color: #fff;
}
.ay:focus-visible {
  background-color: #5c636a;
}
.az:focus-visible {
  box-shadow: 0 0 0 0.25rem rgba(130, 138, 145, 0.5);
}
.ba:focus-visible {
  outline-width: 0;
}
.bb:focus-visible {
  border-top-color: #565e64;
}
.bc:focus-visible {
  border-right-color: #565e64;
}
.bd:focus-visible {
  border-bottom-color: #565e64;
}
.be:focus-visible {
  border-left-color: #565e64;
}
.bm:focus {
  box-shadow: 0 0 0 0.25rem rgba(108, 117, 125, 0.5);
}
.ac:active {
  color: #fff;
}
.ae:active {
  background-color: #6c757d;
}
.af:active {
  background-image: none;
}
.ag:active {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.ah:active {
  outline-width: 0;
}
.ai:active {
  border-top-color: #6c757d;
}
.aj:active {
  border-right-color: #6c757d;
}
.ak:active {
  border-bottom-color: #6c757d;
}
.al:active {
  border-left-color: #6c757d;
}


<button
  data-testid="input-group-btn"
  class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp bq br bs bt bu bv bw bx by d e h i j k l m n o p q r s t u v w x y z a b c d e f g h i j k l input-group-btn"
>
  Group Button</button
>;
"
`;
